src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6740,8 +6828,79 @@ private static bool ReportRefOrOutArgument(AnalyzedArguments analyzedArguments,
             return false;
         }
 
+        /// <summary>
+        /// PROTOTYPE: remove this method before shipping (this is for demo purposes only)
+        /// </summary>
+        private BoundExpression GetPrototypeExtensionIndexderOpt(ExpressionSyntax syntax, BoundExpression receiver, AnalyzedArguments analyzedArguments, DiagnosticBag diagnostics)
```

Comment:
>GetPrototypeExtensionIndexderOpt [](start = 32, length = 32)

@cston @AlekseyTS this is the workaround for demoing the feature. Mainly I want to make sure I'm not introducing a subtle runtime crash for the unconventional overload resolution introduced here.